dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js

```diff
@@ -0,0 +1,97 @@
+//Arrays
+//manifesting answswering all questions this time
+//practice question: Remove duplicates
+var removeDuplicates = function(nums) {
+    let count = 0;
+    //variable to store the number of unique elements ie k
+    for(i=0;i<nums.length;i++){
+        if(nums[i] != nums[i+1]){
+           nums[count] = nums[i];
+        count++
+           }
+    }
+    return count;
+};
+//runtime 68ms space 44.9mb
+//1. Contains duplicate
+//we go through array ~ twice runtime 7366ms space 48.9mbs
+//time complexity  ..... space? 
+var containsDuplicate = function(nums) {
+    for (let i = 0; i < nums.length; i++) {
+        for (let j = 0; j < i;j++) { 
+            //nums.length doesnt compute at this step; instead let j follow in the steps of i
+            if (nums[i] == nums[j]) {
+                //check value of element at those two indices
+                return true;
+            }
+        }
+    }
+    return false;
+};
+
+//2.Binary Search
+// time is big O n- Iterate through once. so depends onm length of array
+//space? 
+//run time  52ms ; spce 45.3mb
+//can i use contains/ is in??
+//other option indexOf
+var search = function(nums, target) {
```

Comment:
The question requires that `You must write an algorithm with O(log n) runtime complexity.` This solution does not satisfy that requirement even though it works. You might want to check out how the binary search algorithm works. I will approve anyways

Reply:
Thanks. I'll look into it as soon as possible.

Reply:
Changes made. I understand now. Thank you.